Tribler/Core/Utilities/configparser.py

```diff
@@ -15,59 +15,58 @@ class CallbackConfigParser(RawConfigParser):
     def __init__(self, *args, **kwargs):
         RawConfigParser.__init__(self, *args, **kwargs)
         self.callback = None
-        self.lock = RLock()
 
+    @blocking_call_on_reactor_thread
```

Comment:
Removing the lock and replacing it with a blocking call is worse than before. I would revert these changes.